### PR TITLE
ARROW-15951: [CI][Python] "Test wheel" step successful despite test error

### PR DESCRIPTION
**Pull request by assignUser**
With no explicit exit code `exit /b` returns 0. This made the the step never fail.
The failing test is only for demonstration and will of course be removed prior to merging this PR.    
I also checked that the same issue isn't happening anywhere else.

### CODE REVIEW

**Comment by assignUser** [comment]
@github-actions crossbow submit wheel-windows-cp310-amd64

File: ci/scripts/python_wheel_windows_test.bat [hunk 1]
```diff
@@ -35,21 +35,21 @@ set ARROW_TEST_DATA=C:\arrow\testing\data
...
+python -c "import pyarrow.csv" || exit /B 1
+python -c "import pyarrow.dataset" || exit /B 1
+python -c "import pyarrow.flight" || exit /B 1
+python -c "import pyarrow.fs" || exit /B 1
+python -c "import pyarrow.json" || exit /B 1
+python -c "import pyarrow.parquet" || exit /B 1
 
 @REM Execute unittest
-pytest -r s --pyargs pyarrow || exit /B
+pytest -r s --pyargs pyarrow || exit /B 1
```
Comment (assignUser) [inline ci/scripts/python_wheel_windows_test.bat:55]: This would probably also have worked:
```suggestion
pytest -r s --pyargs pyarrow
```

**Comment by assignUser** [comment]
@github-actions crossbow submit wheel-windows-cp310-amd64

**Comment by assignUser** [comment]
@kszucs could you take a look at this?

**Review by kszucs — APPROVED** [review]
Nice! Thanks @assignUser for fixing it! (which means making it properly fail in this context)

**Pull request closed without merging**